targets/targets.json

```diff
@@ -583,10 +582,6 @@
             "WATCHDOG",
             "RESET_REASON"
         ],
-        "release_versions": [
```

Comment:
was this already somehow deprecated and we dont use it ? I recall tools relied on this information

Reply:
It will not be used in Mbed 6. More information coming in the Mbed 2 target porting guide. Currently it only really matters for Mbed 2 only targets.

Reply:
OK, I havent seen that update. I assume the clean up will come for all targets (just removing release version).

cc @MarceloSalazar

Reply:
How come you did not override suported_c_libs to build with Microlib?

Reply:
> How come you did not need to override supported_c_libs to build with Microlib?

The following was specified:
```
For all supported Mbed 5 targets identified in previous step:

* Move uARM scatter file (2-region memory model) to ARM toolchain if the ARM scatter file 
  does not supoprt 2-region memory model
* Remove uARM from targets.json
* Remove  TOOLCHAIN_ARM_MICRO directory
* Validate by running greentea tests for ARM toolchain in both RTOS and bare metal mode
```

There was not mention to ensure it builds with Microlib. If that is now also a requirement then I can also ensure it does by overriding `supported_c_libs`.

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/b23c59db1c5e2966d8b82d052bde787178cb270f..58c8d622bb75d5fc870047eb7ff9a94ec3d8ae9d).